distribution/sdk/sample-mpegts-streamgenerator/src/main/java/org/codice/alliance/distribution/sdk/video/stream/mpegts/MpegTsUdpClient.java

```diff
@@ -0,0 +1,305 @@
+        } finally {
+            // Shut down the event loop to terminate all threads.
+            eventLoopGroup.shutdownGracefully();
+        }
+
+        LOGGER.info("count = " + count.get());
+    }
+
+    private static CommandLine getFFmpegInfoCommand(final String videoFilePath) {
+        final String bundledFFmpegBinaryPath = getBundledFFmpegBinaryPath();
+        File file = new File("target/ffmpeg/" + bundledFFmpegBinaryPath);
+        return new CommandLine(file.getAbsolutePath()).addArgument(SUPPRESS_PRINTING_BANNER_FLAG)
+                .addArgument(INPUT_FILE_FLAG)
+                .addArgument(videoFilePath, HANDLE_QUOTING);
+    }
+
+    private static String getBundledFFmpegBinaryPath() {
+        if (SystemUtils.IS_OS_LINUX) {
+            return "linux/ffmpeg";
+        } else if (SystemUtils.IS_OS_MAC) {
+            return "osx/ffmpeg";
+        } else if (SystemUtils.IS_OS_SOLARIS) {
+            return "solaris/ffmpeg";
+        } else if (SystemUtils.IS_OS_WINDOWS) {
+            return "windows/ffmpeg.exe";
+        } else {
+            throw new IllegalStateException("OS is not Linux, Mac, Solaris, or Windows."
+                    + " No FFmpeg binary is available for this OS, so the plugin will not work.");
+        }
+    }
+
+    private static Duration getVideoDuration(final String videoFilePath) {
+        String utfOutputStream;
+        try (final ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
+            final PumpStreamHandler streamHandler = new PumpStreamHandler(outputStream);
+            final CommandLine command = getFFmpegInfoCommand(videoFilePath);
+            final DefaultExecuteResultHandler resultHandler = executeFFmpeg(command,
+                    3,
+                    streamHandler);
+            resultHandler.waitFor();
+            utfOutputStream = outputStream.toString(StandardCharsets.UTF_8.name());
+            return parseVideoDuration(utfOutputStream);
+        } catch (InterruptedException e) {
+            logErrorMessage(String.format("Thread interrupted when executing ffmpeg command. %s",
+                    e), false);
+            return null;
+        } catch (UnsupportedEncodingException e) {
+            logErrorMessage(String.format("Unsupported encoding in ffmpeg output. %s", e), false);
+            return null;
+        } catch (IllegalArgumentException e) {
+            logErrorMessage(String.format("Unable to parse video duration. %s", e), false);
+            return null;
+        } catch (IOException | IllegalStateException e) {
+            logErrorMessage(String.format("Unable to execute ffmpeg command. %s", e), false);
+            return null;
+        }
+    }
+
+    private static Duration parseVideoDuration(final String ffmpegOutput)
+            throws IllegalArgumentException {
+        final Pattern pattern = Pattern.compile("Duration: \\d\\d:\\d\\d:\\d\\d\\.\\d+");
```

Comment:
While 100 hours would be possible to stream, it's unlikely that an FMV clip will be this long